coordinator/src/test/scala/com/socrata/datacoordinator/service/collocation/CoordinatedCollocatorTest.scala

```diff
@@ -299,6 +322,67 @@ class CoordinatedCollocatorTest extends FunSuite with Matchers with MockFactory
     }
   }
 
+  // collocate: alpha.1 (store1A, store3A) with bravo.2 (store5A, store7A)
+  //
+  // moves:
+  //   bravo.2: store5A, store7A -> store1A, store3A
+  //
+  test("explainCollocation for one default store group and a collocated dataset and a non-collocated dataset should move the non-collocated dataset twice") {
+    withMocks(Set(storeGroupA), { coordinator =>
+      (coordinator.secondaryGroups _).expects().returns(Map(storeGroupA -> groupConfig(2, storesGroupA)))
+
+      (coordinator.secondaryMoveJobs _).expects(storeGroupA, alpha1).once.returns(moveJobResultEmpty)
+      (coordinator.secondariesOfDataset _).expects(alpha1).once.returns(secondaries(store1A, store3A))
+
+      (coordinator.secondaryMoveJobs _).expects(storeGroupA, bravo2).once.returns(moveJobResultEmpty)
+      (coordinator.secondariesOfDataset _).expects(bravo2).once.returns(secondaries(store5A, store7A))
+
+      expectCDOIAlpha1WithCollocationsComplex(coordinator) // group for alpha.1
+      expectCDOIDatasetsWithNoCollocations(coordinator, Set(bravo2)) // group for bravo.2
+    }) { case (collocator, _) =>
+      val result = collocator.explainCollocation(storeGroupA, request(Seq((alpha1, bravo2)))).right.get
+      result.id should be (None)
+      result.status should be (Approved)
+      result.message should be (Approved.message)
+      result.cost should be (Cost(2))
+
+      result.moves.size should be (2)
```

Comment:
It might be worth verifying the specific moves instead of the general shape. Currently something moving to a bogus `storeIdTo` would pass.

Reply:
Yeah, problem is I have some random generators... I can fix it though

Reply:
So I don't know exactly what the moves may be, but I can check that the destination stores are in the correct subset